controllers/workspace/devworkspace_controller.go

```diff
@@ -330,6 +337,15 @@ func (r *DevWorkspaceReconciler) Reconcile(req ctrl.Request) (reconcileResult ct
 	return reconcile.Result{}, nil
 }
 
+func cleanupWorkspace(ctx context.Context, workspace *devworkspace.DevWorkspace, client client.Client, returnError error) (reconcile.Result, error) {
+	// Clean up cluster deployment before starting job to avoid mounting issues for RWO PVCs
+	err := provision.ScaleDeploymentToZero(ctx, workspace, client)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	return reconcile.Result{}, returnError
```

Comment:
Not sure i really like this idea of passing in a previous error so we can return it if there was no other error. Is there a better way?

Reply:
fixed!